### PR TITLE
Improve error message for typeerrors

### LINKED ISSUE #730: Ambiguous error - Error: 'Property' must be of type: string
Hi,

I have an issue with the error messages. I am getting the following error:

Error: 'Property' must be of type: string

I obviously have a problem with my data but I have no clue _which_ property is being provided the wrong data. Is there anyway the message could be updated to indicate where the problem lies? Or, if I'm just being dumb, point me to where I can see what property is having issues. 

Thanks,

Sam

### PR DESCRIPTION
**Pull request by kristiandupont**
This resolves #730

### CODE REVIEW

**Comment by kristiandupont** [comment]
This only resolves the issue partly. #787 covers the rest, which requires changes to object store. I think we could merge this one for the current release and make the full solution after that.

File: tests/js/realm-tests.js [hunk 1]
```diff
@@ -866,4 +866,21 @@ module.exports = {
...
+        var realm = new Realm({schema: [schemas.PersonObject]});
+        
+        TestCase.assertThrowsException(function() {
+            realm.write(function () {
+                var p1 = realm.create('PersonObject', { name: 'Ari', age: 10 });
+                p1.age = "Ten";
+            });
+        }, new Error("PersonObject.age must be of type: number"));
+
+        //TestCase.assertThrowsException(function() {
```
Comment (alazier) [inline tests/js/realm-tests.js]: do we still want this?

Comment (kristiandupont) [inline tests/js/realm-tests.js]: You are right, I should remove it and reintroduce it when we solve #787.

**Comment by kristiandupont** [comment]
@fealebenpae I agree that `TypeError` would be better but we aren't using specialized error classes anywhere and introducing them here seems out of scope for this PR.

**Comment by jbrower95** [comment]
is this in 0.15?

**Comment by kristiandupont** [comment]
@jbrower95 yes, 0.15.4.

**Pull request merged**